### PR TITLE
Synchronization of variables from server to client inside modules

### PR DESCRIPTION
**Pull request by Jacob-Mango**
Merge after #72 

```csharp
[CF_RegisterModule(SomeModule)]
class SomeModule : CF_ModuleWorld
{
	private float m_ModuleTime;
	private float m_ModuleTimeSynch;
	override void OnInit()
	{
		super.OnInit();
		RegisterNetSyncVariable("m_ModuleTimeSynch");
		EnableUpdate();
	}
	override void OnVariablesSynchronized(Class sender, CF_EventArgs args)
	{
		super.OnVariablesSynchronized(sender, args);
		
		Print("The client module is at " + m_ModuleTime);
		Print("The server module is at " + m_ModuleTimeSynch);
	}
	override void OnUpdate(float timeslice)
	{
		m_ModuleTime += timeslice;
		if (GetGame().IsServer())
		{
			m_ModuleTimeSynch += timeslice;
			SetSynchDirty();
		}
	}
};
```

### CODE REVIEW

File: JM/CF/Scripts/4_World/CommunityFramework/Module/JMVariableBinding.c [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+		T variable;
+		if (!ctx.Read(variable)) {
+			return false;
+		}
+		
+		EnScript.SetClassVar(instance, m_VariableName, 0, variable);
+		
+		return true;
+	}
+}
```
Comment (Arkensor) [inline JM/CF/Scripts/4_World/CommunityFramework/Module/JMVariableBinding.c:53]: Missing LB on EOL

Comment (Arkensor) [inline JM/CF/Scripts/4_World/CommunityFramework/Module/JMVariableBinding.c]: Missing LB on EOL

File: JM/CF/Scripts/4_World/CommunityFramework/Module/JMModuleBase.c [hunk 2]
```diff
@@ -72,6 +76,37 @@ class JMModuleBase
...
+
+		rpc.Send(null, JMModuleManager.JM_VARIABLE_UPDATE, true, null);
+	}
+
+	void HandleNetSync(ParamsReadContext ctx)
+	{
+		m_NetworkVariables.Read(rpc);
+
+		OnVariablesSynchronized();
+	}
```
Comment (Arkensor) [inline JM/CF/Scripts/4_World/CommunityFramework/Module/JMModuleBase.c]: Will not compile because copy-paste error. `rpc` must be `ctx`

**Review by Arkensor — COMMENTED** [review]
Please make sure your code is tested. Since you committed something that does not even compile this makes me think this was blind-coded.

**Comment by Jacob-Mango** [comment]
@Arkensor never "completed" the previous review so consider this the notification to re-review

**Review by Arkensor — APPROVED** [review]
If tested and working this looks alright :)

**Pull request merged**